### PR TITLE
Fire events `StartRequest` and `EndRequest`

### LINKED ISSUE #61: Fire events
**Is your feature request related to a problem? Please describe.**

I use https://github.com/itsgoingd/clockwork to debug my app. Clockwork keeps track of all triggered events including parameters. In many use cases it would be nice if the sailor actions were accessible as an event.

**Describe the solution you'd like**

add https://github.com/illuminate/events as a dependency and fire on execute and receivedResponse

### PR DESCRIPTION
**Pull request by simbig**
- [x] Added automated tests
- [x] Documented for all relevant versions
- [x] Updated the changelog

**Changes**

- Add `fireEvent`-method to `EndpointConfig::class`

Resolves #61

### CODE REVIEW

**Comment by simbig** [comment]
resolves: https://github.com/spawnia/sailor/issues/61

**Comment by spawnia** [comment]
> resolves: https://github.com/spawnia/sailor/issues/61

Having this in a comment will not automatically  close the issue. It needs to be in the description.

**Comment by spawnia** [comment]
Missing tests.

File: src/Operation.php [hunk 1]
```diff
@@ -85,11 +87,16 @@ protected static function fetchResponse(array $args): Response
...
 
+        $endpointConfig = Configuration::endpoint(static::config(), static::endpoint());
+
         $client = self::$clients[static::class]
-            ?? Configuration::endpoint(static::config(), static::endpoint())
-                ->makeClient();
+            ?? $endpointConfig->makeClient();
+
+        $endpointConfig->fireEvent(new StartRequest(static::document(), $variables));
+        $response = $client->request(static::document(), $variables);
```
Comment (spawnia) [inline src/Operation.php:96]:
```suggestion
        $document = static::document();

        $endpointConfig->fireEvent(new StartRequest($document, $variables));
        $response = $client->request($document, $variables);
```

**Pull request merged**